### PR TITLE
Add docs for creating a plugin

### PR DESCRIPTION
**Pull request by sindresorhus**
Initial plugin docs. We can improve this over time based on feedback when we get more people creating plugins.

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Plugins
+
+The Kap plugin system lets you create custom share targets that appear in the editor export menu. You could for example create a plugin to be able to share a screen recording on YouTube.
```
Comment (CodeTheory) [inline docs/plugins.md]: `be able to` is un-necessary in the sentence, I think.

**Comment by vadimdemedes** [comment]
Great stuff, very detailed! No questions left after I finished reading ;)

File: docs/plugins.md (hunk 1 above)
Comment (sindresorhus) [inline docs/plugins.md]: Done

**Comment by sindresorhus** [comment]
@kevva Good catch. Fixed.

File: docs/plugins.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+- Your package must be named with the `kap-` prefix. For example `kap-giphy`.
+- You must have the `kap-plugin` keyword in package.json. Add additional relevant keywords to improve discovery.
+- The `"description"` in package.json should succinctly describe what you can do with it. For example: `Share GIFs on GIPHY`. Not something like this: `Kap plugin that uploads GIFs to GIPHY`.
+- Use `context.setProgress()` whenever possible to keep the user updated on what's happening.
+- The readme should follow the style of [`kap-giphy`](https://github.com/wulkano/kap-giphy).
+
+## Development
+
+When you develop a plugin it’s useful to be able to try it out in Kap. In the directory of your plugin, run `$ npm link`, go to `~/Library/Application Support/Kap/plugins` and run `$ npm plugin-name `, and then add `"plugin-name": "latest"` to the `"dependencies"` in the package.json there. Your plugin should now be shown in Kap.
```
Comment (kevva) [inline docs/plugins.md]: Missed `link` between `npm` and `plugin-name` :).

File: docs/plugins.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+- `context.config`: Get and set config for you plugin. It’s an instance of [`electron-config`](https://github.com/sindresorhus/electron-config#instance).
+-  `context.request()`: Do a network request, like uploading. It’s a wrapper around [`got`](https://github.com/sindresorhus/got).
+- `context.copyToClipboard(text)`: Copy text to the clipboard. If you for example copy a link to the uploaded recording to the clipboard, don’t forget to `context.notify()` the user about it.
+- `context.notify(text)`: Show a notification.
+- `context.setProgress(text, percentage)`: Update progress information in the Kap export window. Use this whenever you have long-running jobs, like uploading. The `percentage` should be a number between `0` and `1`.
+- `context.openConfigFile()`: Open the plugin config file in the user’s editor.
+- `context.cancel()`: Indicate that the plugin operation canceled for some reason. [Example.](https://github.com/wulkano/kap/blob/efc32d12f381615c9fcfc41065d9c2ee200e8975/app/src/main/save-file-service.js#L28-L31) This closes the Kap export window. If the cancelation was not the result of a user gesture, use `context.notify()` to inform the user why it was canceled.
+
```
Comment (kevva) [inline docs/plugins.md]: The `.filePath()` method is missing.

**Pull request merged**